### PR TITLE
Position control zero position bugfixes and features

### PR DESCRIPTION
**Pull request by eSqadron**
1. Bugfix - quick movement changes forward/backward caused position "zero" to move
   * direction calculation moved to timer interrupt
   * enc interrupt accumulator separated to forward interrupts and backward interrupts
2. Feature added - reset zero to current position
3. Bugfix - keeping position zero (added calculation if negative tolerance for keeping position is below zero)
4. Maitnance - position control universal max position (full spin - 360 deg) moved to define

### CODE REVIEW

File: BrushedMotorControllerFirmware/include/driver.c [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 #define PINS_PER_CHANNEL 2
 
+#define FULL_SPIN_DEGREES (360u * CONFIG_POSITION_CONTROL_MODIFIER)
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/driver.c:17]: maybe DEGREES_PER_ROTATION ?

File: BrushedMotorControllerFirmware/include/driver.c [hunk 2]
```diff
@@ -109,33 +112,54 @@ void enter_boot(void)
...
 		}
-		drv_chnls[chnl].old_count_cycles = drv_chnls[chnl].count_cycles;
-		enum MotorDirection dir;
+
+		if (drv_chnls[chnl].count_cycles_bck > drv_chnls[chnl].old_count_cycles_bck) {
+			diff_bck = drv_chnls[chnl].count_cycles_bck -
+				   drv_chnls[chnl].old_count_cycles_bck;
+		} else {
+			diff_bck = 0;
+		}
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/driver.c:135]: would be nice to just count the cycles and make a diff as negative numbers as well. Is there something blocking us from doing that?

File: BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c [hunk 3]
```diff
@@ -93,6 +93,22 @@ static int cmd_position(const struct shell *shell, size_t argc, char *argv[])
...
 }
 
+static int cmd_position_zero(const struct shell *shell, size_t argc, char *argv[])
+{
+	return_codes_t ret;
+
+	ret = position_reset_zero(get_relevant_channel());
+
+	if (ret == SUCCESS) {
+		shell_fprintf(shell, SHELL_NORMAL, "Position reset succesfulll\n");
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c:103]:
```suggestion
		shell_fprintf(shell, SHELL_NORMAL, "Position reset succesfull\n");
```

**Review by ghost — APPROVED** [review]
few nits/ improvements. can be done in next PRs

File: BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c (hunk 3 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c:103]: Already covered in https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/11

File: BrushedMotorControllerFirmware/include/driver.c (hunk 1 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/driver.c:17]: Agreed, sounds better, corrected in https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/13

File: BrushedMotorControllerFirmware/include/driver.c (hunk 2 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/include/driver.c:135]: actually nothing is blocking us, solved in https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/13, https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/commit/bf46bb1479fa425859f9c62d8ad4dabc6c4d14a3

**Pull request merged**